omod/src/test/java/org/openmrs/module/metadatamapping/web/rest/MetadataTermMappingResourceOperationTest.java

```diff
@@ -0,0 +1,277 @@
+		assertEquals("Test Term Mapping 123", PropertyUtils.getProperty(getResponseData, "name"));
+		Object metadataSource = getResponseData.get("metadataSource");
+		assertEquals("df29a160-0add-4598-8ac2-b11a9eb3cdb8", PropertyUtils.getProperty(metadataSource, "uuid"));
+		
+		Object mappedObject = getResponseData.get("mappedObject");
+		assertEquals("org.openmrs.Drug", PropertyUtils.getProperty(mappedObject, "className"));
+		assertEquals("3cfcf118-931c-46f7-8ff6-7b876f0d4202", PropertyUtils.getProperty(mappedObject, "uuid"));
+	}
+	
+	@Test
+	public void update_shouldPersist() throws Exception {
+		// given
+		SimpleObject postData = new SimpleObject().add("description", "This is the new term mapping description").add(
+		    "mappedObject",
+		    new SimpleObject().add("className", "org.openmrs.Drug").add("uuid", "3cfcf118-931c-46f7-8ff6-7b876f0d4202"));
+		
+		// when
+		MockHttpServletResponse postResponse = handle(newPostRequest(getURI() + "/" + getUuid(), postData));
+		
+		// then
+		assertNotNull(postResponse);
+		
+		SimpleObject getResponseData = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+		assertNotNull(getResponseData);
+		assertEquals("This is the new term mapping description", PropertyUtils.getProperty(getResponseData, "description"));
+		
+		Object mappedObject = getResponseData.get("mappedObject");
+		assertEquals("org.openmrs.Drug", PropertyUtils.getProperty(mappedObject, "className"));
+		assertEquals("3cfcf118-931c-46f7-8ff6-7b876f0d4202", PropertyUtils.getProperty(mappedObject, "uuid"));
+	}
+	
+	@Test
+	public void delete_shouldRetire() throws Exception {
+		// given
+		SimpleObject unRetiredTermMapping = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+		assertEquals(Boolean.FALSE, PropertyUtils.getProperty(unRetiredTermMapping, "retired"));
+		
+		// when
+		MockHttpServletResponse deleteResponse = handle(newDeleteRequest(getURI() + "/" + getUuid()));
+		
+		// then
+		assertNotNull(deleteResponse);
+		
+		SimpleObject retiredTermMapping = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+		assertEquals(Boolean.TRUE, PropertyUtils.getProperty(retiredTermMapping, "retired"));
+	}
+	
+	@Test
+	public void getSingle_shouldGetWithCodeAndSourceUuid() throws Exception {
+		// given
+		// test data
+		
+		// when
+		MockHttpServletRequest request = newGetRequest(getURI() + "/drug-tri");
+		request.setParameter("sourceUuid", "9cace0bd-6f2a-4cc3-a26d-6fa292f1f2c1");
+		SimpleObject getResponseData = deserialize(handle(request));
+		
+		// then
+		assertNotNull(getResponseData);
+		assertEquals("Triomune-30", PropertyUtils.getProperty(getResponseData, "name"));
+	}
+	
+	@Test
+	public void getSingle_shouldGetWithCodeAndSourceName() throws Exception {
+		// given
+		// test data
+		
+		// when
+		MockHttpServletRequest request = newGetRequest(getURI() + "/mdt-xan");
+		request.setParameter("sourceName", "Integration Test Metadata Source 2");
```

Comment:
We typically use ?source={uuid} or ?source={name} for searching of both uuid and name. Would that work here?